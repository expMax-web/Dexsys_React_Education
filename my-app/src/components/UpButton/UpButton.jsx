import styles from "./UpButton.module.css";

export const UpButton = ({ children, scrollToTop }) => {
  return (
    <button onClick={scrollToTop} className={styles.ButtonUp}>
      {children}
    </button>
  );
};
