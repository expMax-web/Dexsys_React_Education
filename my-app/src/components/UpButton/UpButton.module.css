.ButtonUp {
  position: fixed;
  bottom: 50px;
  right: 80px;
  text-align: center;
  transform: translate(-50%, -50%);
  width: 60px;
  height: 60px;
  font-size: 20px;
  color: #fff;
  text-transform: uppercase;
  text-decoration: none;
  transition: 0.5s;
  background-color: #86bcfc;
  border: 4px solid #fff;
  border-radius: 50%;
  overflow: hidden;
  opacity: 0.6;
}

.ButtonUp:hover {
  box-shadow: 0px 0px 13px 4px rgba(34, 60, 80, 0.2);
  opacity: 1;
}

.ButtonUp_dark {
  position: fixed;
  bottom: 50px;
  right: 80px;
  text-align: center;
  transform: translate(-50%, -50%);
  width: 60px;
  height: 60px;
  font-size: 20px;
  color: #fff;
  text-transform: uppercase;
  text-decoration: none;
  transition: 0.5s;
  background-color: #3a3b3b;
  border: 4px solid #fff;
  border-radius: 50%;
  overflow: hidden;
  opacity: 0.6;
}

.ButtonUp_dark:hover {
  box-shadow: 0px 0px 13px 4px rgba(34, 60, 80, 0.2);
  opacity: 1;
}
