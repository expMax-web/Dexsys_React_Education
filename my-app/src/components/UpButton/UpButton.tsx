import React, { useContext } from "react";
import ThemeContext from "../../context";

import styles from "./UpButton.module.css";

interface UpButtonProps {
  scrollToTop: () => void;
}

export const UpButton: React.FC<UpButtonProps> = ({
  children,
  scrollToTop,
}) => {
  const { isDark } = useContext(ThemeContext);
  return (
    <button
      onClick={scrollToTop}
      className={isDark ? styles.ButtonUp_dark : styles.ButtonUp}
    >
      {children}
    </button>
  );
};
