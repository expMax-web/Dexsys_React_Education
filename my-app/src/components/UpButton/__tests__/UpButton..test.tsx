import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { UpButton } from "../UpButton";

describe("UpButton test suite", () => {
  test("should be render correctly ", () => {
    const scrollToTop = jest.fn();

    const { container } = render(<UpButton scrollToTop={scrollToTop} />);
    expect(container).toMatchSnapshot();
  });

  test("scrollToTop work correctly ", () => {
    const scrollToTop = jest.fn();
    render(<UpButton scrollToTop={scrollToTop} />);
    userEvent.click(screen.getByRole("button"));

    expect(scrollToTop).toHaveBeenCalledTimes(1);
  });
});
