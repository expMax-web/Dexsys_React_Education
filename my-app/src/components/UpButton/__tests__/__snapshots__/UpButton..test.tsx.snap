// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton test suite should be render correctly  1`] = `
<div>
  <button
    class="ButtonUp"
  />
</div>
`;

exports[`UpButton test suite scrollToTop work correctly  1`] = `
<div>
  <button
    class="ButtonUp"
  />
</div>
`;

exports[`UpButton test suite should be render correctly  1`] = `
<div>
  <button
    class="ButtonUp"
  />
</div>
`;

exports[`UpButton test suite should be render correctly  2`] = `
<div>
  <button
    class="ButtonUp"
  />
</div>
`;
